### ISSUE TITLE
Incorrect frustum culling after setting custom AABB

### ISSUE THREAD

**Issue by Redwarx008** [godotengine/godot#67526]
### Godot version

4.0B3, 3.51

### System information

Win10 Vulkan/GLES3 Intel(R) UHD Graphics 750

### Issue description

I made a script to generate the block terrain. Modify height in vertex shader. So I need to give each chunked mesh a custom AABB.
Chunk inherits from MeshInstance, so I call the SetCustomAabb method. And then I draw AABB cube with GetTransformedAabb method, seems didn't get the custom AABB. But after that the frustum culling seem not right。
Here are the incorrect culls. The orange line is the AABB cube obtained by GetTransformedAabb()
![图片](https://user-images.githubusercontent.com/50837890/196135282-2b14ac22-b5e7-4eb0-bfa5-558eeedc8649.png)
I drew the custom AABB and it looks correct.
![图片](https://user-images.githubusercontent.com/50837890/196137201-cc2791b5-f1e7-43f1-839c-13419b33c1f5.png)

### Steps to reproduce

see test project

### Minimal reproduction project
[MinTest.zip](https://github.com/godotengine/godot/files/9825802/MinTest.zip)

[GDSMinTest.zip](https://github.com/godotengine/godot/files/9825791/GDSMinTest.zip)

**Comment by clayjohn** [comment]
Could you perhaps provide an MRP that doesn't require C# builds? It would be helpful for contributors who don't use C# and would rule out this being a C#-related issue

**Comment by Redwarx008** [comment]
I tried to rewrite it with GDScript and nothing changed.The GDScript source file has been uploaded

**Comment by Redwarx008** [comment]
I don't know what's happened, in fact, there are a lot of primitive indices even though there's nothing in the viewport.

**Comment by clayjohn** [comment]
How are you defining ``GetTransformedAABB``? In the GDscript sample you rely on ``get_transformed_aabb()`` which doesn't exist so the code won't run.

**Comment by Redwarx008** [comment]
The half size offset is just create mesh, and the position in AABB's constructor should be in world coordinates.That should not be the cause of the problem.
A recent test found that I used the following method to call each frame and got the correct cull.
``` c#
RenderingServer.MeshSetCustomAabb(GetBase(), CustomAABB);
```
And it's the same as getting the auto-generated AABB of the mesh without using the vertex offset in the shader.
But changing to **set_custom_aabb()** or **RenderingServer.InstanceSetCustomAabb()** would still give the wrong result.
New test project has been updated
